config/routes.rb

```diff
@@ -3,10 +3,9 @@
 Rails.application.routes.draw do
   # If the DOMAIN env var is present, and the request doesn't come from that
   # hostname, redirect us to the canonical hostname with the path and query string present
-  if ENV["DOMAIN"].present?
-    hostname = URI(ENV["DOMAIN"]).hostname
-    constraints(host: Regexp.new("^(?!#{Regexp.escape(hostname)})")) do
-      match "/(*path)" => redirect(host: hostname), :via => [:all]
+  if ENV["CANONICAL_HOSTNAME"].present?
+    constraints(host: Regexp.new("^(?!#{Regexp.escape(ENV["CANONICAL_HOSTNAME"])})")) do
```

Comment:
Cripes! I think I understand this!

If the hostname is not the canonical one (www.report-official-development-assistance.service.gov.uk) we redirect to it for all paths and transports – is that right?

Reply:
Yep, that's right 👍

Reply:
This feels like logic that I'm more used to putting in nginx config, or a similar "infrastructure" location, rather than in the app -- is there somewhere more appropriate in our infrastructure for dealing with host-based redirects?

Reply:
I don't think we have that luxury, given we're containerised.